File: app-java/src/main/java/net/mEmoZz/yts/java/ui/splash/SplashPresenter.java
package net.mEmoZz.yts.java.ui.splash;

import net.mEmoZz.yts.java.ui.base.BaseInteractor;

/**
 * Authored by Mohamed Fathy on 08 Mar, 2018.
 * Contact: muhamed.gendy@gmail.com
 */

class SplashPresenter implements SplashContract.Presenter {

  private SplashContract.View view;

  @Override public void onAttach(SplashContract.View view, BaseInteractor interactor) {
    this.view = view;
    this.view.setPresenter(this);
    this.view.openMainActivity();
  }

  @Override public void onDestroy() {
    view = null;
  }
}

File: app-kotlin/src/main/java/net/mEmoZz/yts/kotlin/ui/splash/SplashPresenter.kt
package net.mEmoZz.yts.kotlin.ui.splash

import net.mEmoZz.yts.kotlin.ui.splash.SplashContract.View

/**
 * Authored by Mohamed Fathy on 08 Mar, 2018.
 * Contact: muhamed.gendy@gmail.com
 */

class SplashPresenter : SplashContract.Presenter {

  private var view: SplashContract.View? = null

  override fun onAttach(view: View) {
    this.view = view
    this.view!!.setPresenter(this)
    this.view!!.openMainActivity()
  }

  override fun onDestroy() {
    view = null
  }
}
